Assets/MixedRealityToolkit-Examples/Demos/Boundary/Scripts/BoundaryVisualizer.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using UnityEngine;
+using UnityEngine.Experimental.XR;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    /// <summary>
+    /// Demo class to show different ways of using the boundary API.
+    /// </summary>
+    public class BoundaryVisualizer : MonoBehaviour
+    {
+        /// <summary>
+        /// The material used to display indicators that are within the boundary geometry.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("Material used to display indicators that are within the boundary geometry.")]
+        private Material boundsMaterial = null;
+
+        /// <summary>
+        /// The material used to display indicators that are outside of the boundary geometry.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("Material used to display indicators that are outside of the boundary geometry.")]
+        private Material outOfBoundsMaterial = null;
+
+        /// <summary>
+        /// The material used to display the inscribed rectangle and the indicators that are within it.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("Material used to display the inscribed rectangle and the indicators that are within it.")]
+        private Material inscribedRectangleMaterial = null;
+
+        /// <summary>
+        /// Boundary system implementation.
+        /// </summary
+        private IMixedRealityBoundarySystem boundaryManager = null;
+        private IMixedRealityBoundarySystem BoundaryManager => boundaryManager ?? (boundaryManager = MixedRealityManager.Instance.GetManager<IMixedRealityBoundarySystem>());
+
+        private void Start()
+        {
+            AddQuad();
+            AddIndicators();
+        }
+
+        /// <summary>
+        /// Displays the boundary as a quad primitive.
+        /// </summary>
+        private void AddQuad()
+        {
+            // Get the rectangular bounds.
+            Vector2 center = EdgeUtilities.InvalidPoint;
+            float angle = 0f;
+            float width = 0f;
+            float height = 0f;
+            if (!(bool)(BoundaryManager?.TryGetRectangularBoundsParams(out center, out angle, out width, out height)))
```

Comment:
Why do we need to cast to a bool if the return type is a bool?

I'm also seeing a possible invalid operation exception

Reply:
i almost wrote that code originally :)